openmc/filter.py

```diff
@@ -266,35 +215,9 @@ def can_merge(self, other):
 
         """
 
-        if not isinstance(other, Filter):
-            return False
-
-        # Filters must be of the same type
-        if self.type != other.type:
-            return False
-
-        # Distribcell filters cannot have more than one bin
-        if self.type == 'distribcell':
-            return False
+        if type(self) is not type(other): return False
```

Comment:
We actually have to use `type(...) is` in this case rather than `isinstance(...)` because of inheritance.
```python
f = openmc.EnergyoutFilter(bins)
isinstance(f, openmc.EnergyFilter) # This returns True
type(f) is openmc.EnergyFilter # This returns False
```